src/Shadowdomize/DOMFunctions.ts

```diff
@@ -0,0 +1,226 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+export function getActiveElement(doc: Document): Element | null {
+    let activeElement = doc.activeElement;
+
+    while (activeElement?.shadowRoot?.activeElement) {
+        activeElement = activeElement.shadowRoot.activeElement;
+    }
+
+    return activeElement;
+}
+
+export function nodeContains(
+    node: Node | null | undefined,
+    otherNode: Node | null | undefined
+): boolean {
+    if (!node || !otherNode) {
+        return false;
+    }
+
+    let currentNode: HTMLElement | Node | null | undefined = otherNode;
+
+    while (currentNode) {
+        if (currentNode === node) {
+            return true;
+        }
+
+        if (
+            typeof (currentNode as HTMLSlotElement).assignedElements !==
+                "function" &&
+            (currentNode as HTMLElement).assignedSlot?.parentNode
+        ) {
+            // Element is slotted
+            currentNode = (currentNode as HTMLElement).assignedSlot?.parentNode;
+        } else if (currentNode.nodeType === document.DOCUMENT_FRAGMENT_NODE) {
+            // Element is in shadow root
+            currentNode = (currentNode as ShadowRoot).host;
+        } else {
+            currentNode = currentNode.parentNode;
+        }
+    }
+
+    return false;
+}
+
+export function getParentNode(
+    node: Node | null | undefined
+): ParentNode | null {
+    if (!node) {
+        return null;
+    }
+
+    if (
```

Comment:
We should either add it now or add a `TODO` to handle slotted nodes.

Reply:
Added TODO to index.ts.